Source/Dafny/Verifier/Translator.ClassMembers.cs

```diff
@@ -893,9 +891,11 @@ public partial class Translator {
       };
       var ens = new List<Boogie.Ensures>();
 
-      var proc = new Boogie.Procedure(f.tok, "OverrideCheck$$" + f.FullSanitizedName, new List<Boogie.TypeVariable>(),
+      var name = MethodName(f, MethodTranslationKind.OverrideCheck);
+      var proc = new Boogie.Procedure(f.tok, name, new List<Boogie.TypeVariable>(),
         Util.Concat(Util.Concat(typeInParams, inParams_Heap), inParams), outParams,
         req, mod, ens, etran.TrAttributes(f.Attributes, null));
+      AddVerboseName(proc, f.FullDafnyName, MethodTranslationKind.OverrideCheck);
```

Comment:
I wonder if we could merge the calls to `MethodName` and `AddVerboseName`, since it seems that a call to the first must always be followed by a call to the second

Reply:
(Especially given that we repeat the `MethodTranslationKind`

Reply:
It's a little less clear how to abstract this cleanly. The result of `MethodName` is used only in the call to `new Bpl.Procedure`, but not every call site for `new Bpl.Procedure` has a preceding `MethodName` call.